packages/pn-pa-webapp/src/redux/apiKeys/reducers.ts

```diff
@@ -13,14 +24,28 @@ const apiKeysSlice = createSlice({
   initialState,
   reducers: {
     resetState: () => initialState,
+    setPagination: (state, action: PayloadAction<{ page: number; size: number }>) => {
+      if (state.pagination.size !== action.payload.size) {
+        // reset pagination
+        state.pagination.nextPagesKey = [];
+        state.pagination.lastUpdate = [];
+        state.pagination.moreResult = false;
+      }
+      state.pagination.size = action.payload.size;
+      state.pagination.page = action.payload.page;
+    },
   },
   extraReducers: (builder) => {
     builder.addCase(getApiKeys.fulfilled, (state, action) => {
       state.apiKeys = action.payload;
+      if (action.payload.lastKey && action.payload.lastUpdate) {
+        state.pagination.nextPagesKey.push(action.payload.lastKey);
```

Comment:
here we push into the array of object `{lastKey: ..., lastUpdate: ...}`. pay attention to not push the same object twice

Reply:
changes made into 5eb41f24bacd7862f6c9ac99ae9f9e9396e71a40